tests/manager/api/controller/test_playtime_entries.py

```diff
@@ -1,30 +1,45 @@
 import datetime
 import hashlib
-from unittest.mock import patch
+from unittest.mock import MagicMock, patch
 
-import flask
 import pytest
 from sqlalchemy.exc import IntegrityError
 
 from palace.manager.api.controller.playtime_entries import (
     MISSING_LOAN_IDENTIFIER,
+    PlaytimeEntriesController,
     resolve_loan_identifier,
 )
 from palace.manager.sqlalchemy.model.patron import Loan
 from palace.manager.sqlalchemy.model.time_tracking import PlaytimeEntry
 from palace.manager.sqlalchemy.util import get_one
 from palace.manager.util.datetime_helpers import utc_now
 from palace.manager.util.problem_detail import ProblemDetail
-from tests.fixtures.api_controller import CirculationControllerFixture
+from tests.fixtures.database import DatabaseTransactionFixture
+from tests.fixtures.flask import FlaskAppFixture
 
 
-def date_string(hour=None, minute=None):
-    now = utc_now()
-    if hour is not None:
-        now = now.replace(hour=hour)
-    if minute is not None:
-        now = now.replace(minute=minute)
-    return now.strftime("%Y-%m-%dT%H:%M:00+00:00")
+class PlaytimeEntriesControllerFixture:
+    def __init__(self, db: DatabaseTransactionFixture) -> None:
+        mock_manager = MagicMock()
+        mock_manager._db = db.session
+        self.controller = PlaytimeEntriesController(mock_manager)
+
+    @staticmethod
+    def date_string(hour=None, minute=None):
+        now = utc_now()
+        if hour is not None:
+            now = now.replace(hour=hour)
+        if minute is not None:
+            now = now.replace(minute=minute)
+        return now.strftime("%Y-%m-%dT%H:%M:00+00:00")
+
+
+@pytest.fixture()
+def playtime_entries_controller_fixture(
+    db: DatabaseTransactionFixture,
+) -> PlaytimeEntriesControllerFixture:
+    return PlaytimeEntriesControllerFixture(db)
 
 
 class TestPlaytimeEntriesController:
```

Comment:
All the changes to these tests are so we can use `flask_app_fixture` in the tests, which allows us to pass `patron` and `library` to `test_request_context`. That way we aren't directly manipulating `flask.request` in the tests.